src/StateManager.cpp

```diff
@@ -0,0 +1,17 @@
+#include "StateManager.hpp"
+
+StateManager::StateManager() {
+    currentState = GameState();
+}
+
+void StateManager::update() {
+    currentState.update();
```

Comment:
You might want to add `this->` everywhere for consistency, but that's your choice.
```suggestion
    this->currentState.update();
```